File: swift/problem724.swift
class Solution {
    func pivotIndex(_ nums: [Int]) -> Int {
        let sum = nums.reduce(0,+); var left = 0
        
        for i in 0..<nums.count {
            let tmp = left + nums[i]
            if (left == sum - tmp) {
                return i
            }
            left = tmp
        }
        
        return -1
    }
}

File: python/problem724.py
class Solution:
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        wholesum = sum(nums)
        leftsum = 0
        for i in range(len(nums)):
            
            if wholesum - leftsum - nums[i] == leftsum:
                return i
            leftsum+=nums[i]
        return -1
            
